### ISSUE TITLE
[A11y][iOS]: Voice Control with overlay cannot dismiss an opened dialog with "show number" or "go back"

### ISSUE THREAD

**Issue by ChristianEdwardPadilla** [flutter/flutter#106313]
## Steps to Reproduce

1. Enable Voice Control with Overlay in iOS settings.
2. Trigger a Flutter dialog.
3. Attempt to dismiss the dialog using voice controls.

Expected: Voice commands exist that can dismiss the Flutter dialog.
Actual: Seems to be no way to dismiss the dialog using voice controls.

```dart
OutlinedButton(
          onPressed: () {
            showGeneralDialog<void>(
              context: context,
              barrierColor: Colors.transparent,
              barrierDismissible: true,
              barrierLabel:
                  MaterialLocalizations.of(context).modalBarrierDismissLabel,
              pageBuilder: (_, __, ___) =>
                  const AlertDialog(title: Text('Alert!')),
            );
          },
          child: const Text('Open Dialog'),
        ),
```

ATTACHED VIDEO BELOW:
In the video, observe that the dialog cannot be closed.

The number 1 that appears after the command show numbers is to stop the recording, it's not actually there without the video recording. There is no number on the screen when "show numbers" is issued.

Works in Android, and works in VoiceOver. In VoiceOver, the focus can go to the back (I'm guessing there's a ModalBarrier at the back?) and it's announced as "dismiss".

https://user-images.githubusercontent.com/37954976/174659422-049a364e-f91f-4b1d-87fe-53b018a2732d.MP4

**Comment by danagbemava-nc** [comment]
Issue is reproducible on `stable` and `master`. 

<details>
<summary>video</summary>

https://user-images.githubusercontent.com/88313112/174745343-570f0f46-f6d2-4072-b270-5d7d248c2629.MP4

</details>


<details>
<summary>complete code sample</summary>

```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            OutlinedButton(
              onPressed: () {
                showGeneralDialog<void>(
                  context: context,
                  barrierColor: Colors.transparent,
                  barrierDismissible: true,
                  barrierLabel: MaterialLocalizations.of(context).modalBarrierDismissLabel,
                  pageBuilder: (_, __, ___) => const AlertDialog(title: Text('Alert!')),
                );
              },
              child: const Text('Open Dialog'),
            ),
            OutlinedButton(
              onPressed: () {
                showCupertinoDialog<void>(
                  context: context,
                  barrierDismissible: true,
                  barrierLabel: MaterialLocalizations.of(context).modalBarrierDismissLabel,
                  builder: (BuildContext context) => const CupertinoAlertDialog(title: Text('Alert!')),
                );
              },
              child: const Text('Open Cupertino Dialog'),
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}

```
</details>

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 3.0.2, on macOS 12.4 21F79 darwin-arm, locale en-GB)
    • Flutter version 3.0.2 at /Users/nexus/dev/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cd41fdd495 (13 days ago), 2022-06-08 09:52:13 -0700
    • Engine revision f15f824b57
    • Dart version 2.17.3
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-32, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.68.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.42.0

[✓] Connected device (4 available)
    • Nexus (mobile)     • 00008020-001875E83A38002E            • ios            • iOS 15.5 19F77
    • iPhone 13 (mobile) • DBB48503-41CE-4969-9043-3261726E438A • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-4 (simulator)
    • macOS (desktop)    • macos                                • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 102.0.5005.115

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[✓] Flutter (Channel master, 3.1.0-0.0.pre.1317, on macOS 12.4 21F79 darwin-arm, locale en-GB)
    • Flutter version 3.1.0-0.0.pre.1317 on channel master at /Users/nexus/dev/sdks/flutters
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d4929bc828 (8 hours ago), 2022-06-20 20:09:05 -0400
    • Engine revision ceac23d7a9
    • Dart version 2.18.0 (build 2.18.0-190.0.dev)
    • DevTools version 2.14.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-32, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13E500a
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.68.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.42.0

[✓] Connected device (4 available)
    • Nexus (mobile)     • 00008020-001875E83A38002E            • ios            • iOS 15.5 19F77
    • iPhone 13 (mobile) • DBB48503-41CE-4969-9043-3261726E438A • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-4 (simulator)
    • macOS (desktop)    • macos                                • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 102.0.5005.115

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by xster** [comment]
Marking as P3 to match b/228415209

**Comment by jmagman** [comment]
cc @chunhtai who I believe was also cc'd on that b/

**Comment by khyati82** [comment]
@chunhtai - This is one of customers top accessibility issue.

**Comment by stuartmorgan** [comment]
@chunhtai Is this something you would have time to follow up on, or do we need to find iOS team resources for this?

**Comment by chunhtai** [comment]
My schedule is current quite full. Would be good if we can have help from iOS team. From a quick glance. the show number will only show up if the gesture has role `button`.

The goBack is probably going through the SemanticsAction.dismiss
https://api.flutter.dev/flutter/dart-ui/SemanticsAction/dismiss-constant.html

Would need to check whether there action is missing in the repro.

**Comment by chunhtai** [comment]
looks like macos web is working fine, I am able to use voice control to show number and go back when dialog is opened
